lib/server.js

```diff
@@ -255,6 +255,9 @@ function expiry(block2cache, k) {
 
 CoAPServer.prototype._handle = function(packet, rsinfo) {
 
+  // Make sure this is a request
+  if (packet.code[0] != '0') return;
+
```

Comment:
If the code[0] is not 0, this packet is a response of a previous request from the same socket.